### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by amit-mb**

### CODE REVIEW

**Comment by revathskumar** [comment]
@amit-mb Please rebase this branch with master

File: app/__tests__/popup.test.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    test('on stop button click, should trigger disbable message and updateField', () =>{
+      wrapper.find('button').first().props().onClick()
+      wrapper.find('button').first().props().onClick()
+      expect(updateFieldMock).toBeCalled();
+      expect(MessageService.disableLogging).toBeCalled();
+    });
+  });
+
+  describe('on error', () => {
+    let wrapper = shallow(<Popup {...commonProps() } errorMessage={"Error"}  />)
```
Comment (revathskumar) [inline app/__tests__/popup.test.tsx]: either keep this in `beforeEach` or do in the `test` itself.

File: app/__tests__/popup.test.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+///<reference path='./../../node_modules/@types/jest/index.d.ts'/>
+import * as React from 'react';
+import { configure } from 'enzyme';
+import { shallow } from 'enzyme';
+import * as Adapter from 'enzyme-adapter-react-15';
+import {Popup} from './../popup';
+
+import * as MessageService from '../message_service';
+
+configure({ adapter: new Adapter() });
```
Comment (revathskumar) [inline app/__tests__/popup.test.tsx]: move this to `tests/setup.js` and configure it in jest config `"setupTestFrameworkScriptFile": "<rootDir>/tests/setup.js"`

File: app/__tests__/popup.test.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  describe('on enabled', () => {
+
+    beforeEach(() => {
+      wrapper = shallow(<Popup {...commonProps() } enabled={true} />);
+    });
+
+
+    test('on stop button click, should trigger disbable message and updateField', () =>{
+      wrapper.find('button').first().props().onClick()
+      wrapper.find('button').first().props().onClick()
```
Comment (revathskumar) [inline app/__tests__/popup.test.tsx]: why we are calling this twice?

Comment (revathskumar) [reply app/__tests__/popup.test.tsx]: since we are passing `enabled` as `true` it needs only one click to stop it. else there is something wrong.

Comment (revathskumar) [reply app/__tests__/popup.test.tsx]: Also never call onClick. always `simulate`.

File: app/__tests__/popup.test.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+      expect(MessageService.getRequests).toBeCalled();
+      expect(MessageService.enableLogging).toBeCalled();
+      expect(updateFieldsMock).toBeCalled();
+    });
+
+    test('on clear button click, should trigger clearData and clearField', () => {
+      wrapper.find('.btn-clear').simulate('click');
+      expect(MessageService.clearData).toBeCalled();
+      expect(clearFieldsMock).toBeCalled();
```
Comment (revathskumar) [inline app/__tests__/popup.test.tsx]: Try asset with [toHaveBeenCalledTimes()](https://facebook.github.io/jest/docs/en/expect.html#tohavebeencalledtimesnumber). and see this fails or not?

File: app/__tests__/popup.test.tsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+      expect(wrapper.find('.popup-error-message').text()).toEqual(expect.stringMatching("Error"))
+    });
+  });
+
+  describe('on invalid url', () => {
+    beforeEach(() => {
+      let wrapper = shallow(<Popup {...commonProps() } tabId={-1}  />)
+    });
+    test('should call errorNotify', () => {
+      expect(errorNotifyMock).toBeCalled();
```
Comment (revathskumar) [inline app/__tests__/popup.test.tsx]: Try assert with [toHaveBeenCalledWith](https://facebook.github.io/jest/docs/en/expect.html#tohavebeencalledwitharg1-arg2-)

**Comment by amit-mb** [comment]
[DO NOT MERGE] Some tests failing.

**Pull request merged**